activerecord/test/cases/finder_test.rb

```diff
@@ -693,6 +693,11 @@ def test_take_and_first_and_last_with_integer_should_return_an_array
     assert_kind_of Array, Topic.last(5)
   end
 
+  def test_take_and_first_and_last_with_limit_should_return_same_size
+    topics = Topic.limit(1)
+    assert_equal topics.first(2).size, topics.last(2).size
```

Comment:
`first` with `limit` doesn't work as expected, and inconsistent with `last` is true.
But it is also inconsistent with loaded `first` with `limit`.
Can you more exercise tests like `test_last_on_relation_with_limit_and_offset`?

```ruby
  def test_find_on_relation_with_limit_and_offset
    post = posts("sti_comments")

    comments = post.comments.order(id: :asc)
    assert_equal comments.limit(2).to_a.first, comments.limit(2).first
    assert_equal comments.limit(2).to_a.first(2), comments.limit(2).first(2)
    assert_equal comments.limit(2).to_a.first(3), comments.limit(2).first(3)

    assert_equal comments.offset(2).to_a.first, comments.offset(2).first
    assert_equal comments.offset(2).to_a.first(2), comments.offset(2).first(2)
    assert_equal comments.offset(2).to_a.first(3), comments.offset(2).first(3)

    comments = comments.offset(1)
    assert_equal comments.limit(2).to_a.first, comments.limit(2).first
    assert_equal comments.limit(2).to_a.first(2), comments.limit(2).first(2)
    assert_equal comments.limit(2).to_a.first(3), comments.limit(2).first(3)
  end
```
 
https://github.com/rails/rails/blob/7c69351277edaa47ee395c72d59623162eb8682f/activerecord/test/cases/finder_test.rb#L672-L688

Reply:
No problem, I have added additional tests for compatibility here between `limit(x).first(y)` and `limit(x).to_a.first(y)`, and they all now pass as well.